### DISCUSSION TITLE
Non obvious behaviour of body add_paragraph() after an update of the package

### DISCUSSION THREAD

**Discussion by kapytski** [davidgohel/officer#476] [Q&A]
Hello to all, hello to David!
I've faced with a strange behaviour of body_add_paragraph() after the last update.
When this function is used inside an another function, at an output document it produces the backward direction of added paragraphs.
In a case of using for() - it don't.
It's the code:

```
require(officer)

doc<-read_docx()
foo<-function(i){
  doc<-body_add_par(doc,i)
  doc<-body_add_par(doc,paste0(""))
}
sapply(1:10,foo)
print(doc,"an issue.docx")

doc<-read_docx()
for(i in 1:10){
  doc<-body_add_par(doc,i)
  doc<-body_add_par(doc,paste0(""))
}
print(doc,"not an issue.docx")
```
There are results of using 2 approaches:
![изображение](https://user-images.githubusercontent.com/32036859/223218810-3bdf936f-b6be-4f26-9ab3-01d6a338f528.png)

How to fix it?

**Comment by kapytski** [comment]
Sorry! I've forgotten to point information about my system: Win 10 (21H2), officer 0.6.0

>version
platform       x86_64-w64-mingw32               
arch           x86_64                           
os             mingw32                          
crt            ucrt                             
system         x86_64, mingw32                  
status                                          
major          4                                
minor          2.2                              
year           2022                             
month          10                               
day            31                               
svn rev        83211                            
language       R                                
version.string R version 4.2.2 (2022-10-31 ucrt)
nickname       Innocent and Trusting

**Comment by davidgohel** [comment]
Hello 

This is related to a change/fix that happened here: https://github.com/davidgohel/officer/commit/a6e88f1164792388df678731bcd1cfa0be9f26e9


```
doc<-read_docx()
foo<-function(i){
  doc<-body_add_par(doc,i)
  doc<-body_add_par(doc,paste0("")) # no return for the function
}
sapply(1:10,foo)# results are not assigned
```


Cursor had issues (it was resource consuming and sometimes bugged) and we replaced it with a clean S3 system. Before it was based on an R6 class. So when you were not assigning the result, it was working but only because of a *clunky* implementation and it was persistent because of R6. Assignment has always been required – I am sure other strange things happened because of that.

For this reason, `sapply` is not a good approach, you can't sapply *ggplot* commands, same applies here, that's also why you can `%>%` operation with officer. 

`walk()` and `for()` are the most natural solution to me, but I am sure there are others.  So the second approach is the best to me. 

```
doc<-read_docx() # doc is created
for(i in 1:10){
  doc<-body_add_par(doc,i) # doc is augmented and the result goes in doc
  doc<-body_add_par(doc,paste0(""))  # doc is augmented and the result goes in doc
}
print(doc,"not an issue.docx")  # doc is printed
```

David

**Reply by davidgohel** [reply]
Let me know if unclear so that I can update if necessary

**Reply by kapytski** [reply]
Thank you, David! For the answers and for your package!
